### PR TITLE
Feature/book facets json [WIP]

### PR DESCRIPTION
**Pull request by bwhicks**
Creating this initially as a WIP @thatbudakguy @rlskoeser. The code is 99% there, but I can see a few small changes I'll make if time permits.

It is useable for testing against Vue.js form solutions, but does require a fresh solr core because of a rename.

### CODE REVIEW

**Comment by thatbudakguy** [comment]
It looks like the lines codecov is angry about are parts of the `RangeFacet` logic that we borrowed from Derrida - maybe going forward we can break that off into a small package or something so that it comes with its own tests and we don't have to worry about copy-pasting the tests along with the code? Not sure though since it's so small.

File: winthrop/books/forms.py [hunk 1]
```diff
@@ -55,28 +137,106 @@ class SearchForm(forms.Form):
...
+        'editor_exact': 'editor',
+        'translator_exact': 'translator',
+        'language_exact': 'language',
+        'subject_exact': 'subject',
+        'annotator_exact': 'annotator'
+    }
+
+    # TODO: Should this be a dict? Right now it doesn't need a different name
+    # to map to the fields used in the form.
+    range_facets = ['pub_year']
```
Comment (thatbudakguy) [inline winthrop/books/forms.py]: range fields won't ever have suffixes like "_exact", right? this seems fine to me.

**Comment by rlskoeser** [comment]
@thatbudakguy wrote:
> It looks like the lines codecov is angry about are parts of the RangeFacet logic that we borrowed from Derrida - maybe going forward we can break that off into a small package or something so that it comes with its own tests and we don't have to worry about copy-pasting the tests along with the code? Not sure though since it's so small.

I'm not sure where this belongs yet in terms of a reusable package (maybe part of parasol?), so in the meantime I'm inclined to say we should just copy the tests along with the code.

File: winthrop/books/models.py [hunk 2]
```diff
@@ -287,7 +296,13 @@ def index_data(self):
             'authors': [str(author) for author in self.authors()],
             # first author only, for sorting
             # FIXME: sort on last name first? not an ordered relationship currently
-            'author_exact': str(self.authors().first()) if self.authors().exists() else None,
+            # NOTE: author_exact is a copy field.
+            'author_sort': str(self.authors().first()) if self.authors().exists() else None,
+            'editor_exact': [str(editor) for editor in self.editors()],
```
Comment (rlskoeser) [inline winthrop/books/models.py]: @bwhicks I can see why you did this, but it feels a bit weird to me. How do you feel about switching it to generic fields (editor, translator, subject) that we then copy into the exact fields and the main text search field?

Comment (bwhicks) [reply winthrop/books/models.py]: @rlskoeser That should be a lot cleaner. I'll make the change.

**Pull request merged**